api/v1beta2/gitrepository_types.go

```diff
@@ -29,165 +29,197 @@ const (
-	// If not provided, a default will be used, consult the documentation for your version to find out what those are.
+	// Ignore overrides the set of excluded patterns in the .sourceignore format
+	// (which is the same as .gitignore). If not provided, a default will be used,
+	// consult the documentation for your version to find out what those are.
 	// +optional
 	Ignore *string `json:"ignore,omitempty"`
 
-	// Suspend tells the controller to suspend the reconciliation of this source.
-	// This flag tells the controller to suspend the reconciliation of this source.
+	// Suspend tells the controller to suspend the reconciliation of this
+	// GitRepository.
 	// +optional
 	Suspend bool `json:"suspend,omitempty"`
 
-	// Determines which git client library to use.
-	// Defaults to go-git, valid values are ('go-git', 'libgit2').
+	// GitImplementation specifies which Git client library implementation to
+	// use. Defaults to 'go-git', valid values are ('go-git', 'libgit2').
 	// +kubebuilder:validation:Enum=go-git;libgit2
 	// +kubebuilder:default:=go-git
 	// +optional
 	GitImplementation string `json:"gitImplementation,omitempty"`
 
-	// When enabled, after the clone is created, initializes all submodules within, using their default settings.
+	// RecurseSubmodules enables the initialization of all submodules within
+	// the GitRepository as cloned from the URL, using their default settings.
 	// This option is available only when using the 'go-git' GitImplementation.
 	// +optional
 	RecurseSubmodules bool `json:"recurseSubmodules,omitempty"`
 
-	// Include defines a list of GitRepository resources which artifacts should be included in the artifact produced for
-	// this resource.
+	// Include specifies a list of GitRepository resources which Artifacts
+	// should be included in the Artifact produced for this GitRepository.
 	Include []GitRepositoryInclude `json:"include,omitempty"`
 
-	// AccessFrom defines an Access Control List for allowing cross-namespace references to this object.
+	// AccessFrom specifies an Access Control List for allowing cross-namespace
+	// references to this object.
 	// +optional
 	AccessFrom *acl.AccessFrom `json:"accessFrom,omitempty"`
 }
 
+// GitRepositoryInclude specifies a reference to a GitRepository which Artifact
+// (sub-)contents must be included, and where they should be placed.
+type GitRepositoryInclude struct {
+	// GitRepositoryRef specifies the GitRepository which Artifact contents
+	// must be included.
+	GitRepositoryRef meta.LocalObjectReference `json:"repository"`
+
+	// FromPath specifies the path to copy contents from, defaults to the root
+	// of the Artifact.
+	// +optional
+	FromPath string `json:"fromPath"`
+
+	// ToPath specifies the path to copy contents to, defaults to the name of
+	// the GitRepositoryRef.
+	// +optional
+	ToPath string `json:"toPath"`
+}
+
+// GetFromPath returns the specified FromPath.
```

Comment:
I feel like `func (in *GitRepositoryInclude) FromPath() string` and `func (in *GitRepositoryInclude) ToPath() string` are more intuitive names.

Reply:
Fields and methods can not bear the same name.